portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -0,0 +1,123 @@
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/data")
+public final class DataServlet extends HttpServlet {
+    // Set default to 5
+
+    int maxCommentsPosted = 5;
+	@Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+    
+        // Gather all the user comments and sort them by timestamp
+        Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Task> commentsRecord = new ArrayList<>();
+        int count = 0;
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String username = (String) entity.getProperty("userName");
+            String comment = (String) entity.getProperty("userComment");
+            long timestamp = (long) entity.getProperty("timestamp");    
+            // Limit amount of comments that will be shown to 4
+            count++;    
+            if(count <= maxCommentsPosted){
+                Task userEntry = new Task(id, username, comment, timestamp);
+            	commentsRecord.add(userEntry);
+            }
+            
+    	}
+
+	    // Create commentsRecord object in json form
+        String json = new Gson().toJson(commentsRecord);
+        
+        // Send the JSON as the response
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+
+    }
+
+	/* Receive Any New Inputed Comments from User */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get Input from the Form
+        String userName = request.getParameter("user-name");
+    	String userComment = request.getParameter("user-comment");
+        long timestamp = System.currentTimeMillis();
+
+        // Get max number of comments to show 
+        maxCommentsPosted = getMaxComments(request);
+
+        // Add Input to the Master list of User Comments
+        Entity inputEntity = new Entity("Comments");
+        inputEntity.setProperty("userName", userName);
+        inputEntity.setProperty("userComment", userComment);
+        inputEntity.setProperty("timestamp", timestamp);
+
+        // if statement to ensure empty usernames nor usercomments are put into database
+	    if(inputEntity.getProperty("userName")!= null || inputEntity.getProperty("userComment")!= null){
+        	DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        	datastore.put(inputEntity);
```

Comment:
does datastore.put throw an exception? Please wrap around with try and catch and properly handle.

Reply:
It's not really an exception that it would throw. It's just that whenever this if statement is false, I don't want it to add that empty entity to the datastore.